cou/cli.py

```diff
@@ -42,110 +71,333 @@ def parse_args(args: Any) -> argparse.Namespace:
+    group = subcommand_common_opts_parser.add_mutually_exclusive_group()
+    group.add_argument(
+        "--verbose",
+        "-v",
+        default=0,
+        action="count",
+        dest="verbosity",
+        help="Increase logging verbosity in STDOUT. Repeat the 'v' in the short option "
+        "for more detail. Maximum verbosity is obtained with 4 (or more) "
+        "v's, i.e. -vvvv. \nNote that this doesn't affect the verbosity in logfile, "
+        "which will always have the maximum verbosity.",
+    )
+    group.add_argument(
+        "--quiet",
+        "-q",
+        default=False,
+        action="store_true",
+        dest="quiet",
+        help="Disable output in STDOUT.",
+    )
+
+    # Define options specific to data-plane child command and make them mutually exclusive
+    dp_subparser = argparse.ArgumentParser(add_help=False)
+    dp_upgrade_group = dp_subparser.add_mutually_exclusive_group()
+    dp_upgrade_group.add_argument(
+        "--machine",
+        "-m",
+        action="append",
+        help="Specify machines ids to upgrade.",
+        dest="machines",
+    )
+    dp_upgrade_group.add_argument(
+        "--hostname",
+        "-n",
+        action="append",
+        help="Specify machine hostnames to upgrade.",
+        dest="hostnames",
+    )
+    dp_upgrade_group.add_argument(
+        "--availability-zone",
+        "--az",
+        action="append",
+        help="Specify availability zones to upgrade.",
+        dest="availability_zones",
+    )
+
+    # Arg parser for "cou plan" sub-command
+    plan_parser = subparsers.add_parser(
+        "plan",
+        description="Show the steps for upgrading the cloud to the next release.",
+        help="Show the steps for upgrading the cloud to the next release.",
+        usage="cou plan [options]",
+        parents=[subcommand_common_opts_parser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+
+    # Create control-plane and data-plane child commands to plan partial upgrades
+    plan_subparser = plan_parser.add_subparsers(
+        title="Upgrade group",
+        dest="upgrade-group",
+        help="For more information about a upgrade group, run 'cou plan <upgrade-group>' -h.",
+    )
+    plan_subparser.add_parser(
+        "control-plane",
+        description="Show the steps for upgrading the control-plane components.",
+        help="Show the steps for upgrading the control-plane components.",
+        usage="cou plan control-plane [options]",
+        parents=[subcommand_common_opts_parser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
+
+    plan_subparser.add_parser(
+        "data-plane",
+        description="Show the steps for upgrading the data-plane components.",
+        help="Show the steps for upgrading the data-plane components.",
+        usage="cou plan data-plane [options]",
+        parents=[subcommand_common_opts_parser, dp_subparser],
+        formatter_class=CapitalisedHelpFormatter,
+    )
```

Comment:
we should indicate that steps will not be shown if the control plane hasn't been upgraded